lib/ui/window.dart

```diff
@@ -138,8 +138,42 @@ class FlutterView {
   ///    The value here is equal to the value exposed on [display].
   double get devicePixelRatio => _viewConfiguration.devicePixelRatio;
 
-  /// The dimensions of the rectangle into which the scene rendered in this view
-  /// will be drawn on the screen, in physical pixels.
+  /// The sizing constraints in physical pixels for this view.
+  ///
+  /// The view can take on any [Size] that fulfills these constraints. These
+  /// constraints are typically used by an UI framework as the input for its
+  /// layout algorithm to determine an approrpiate size for the view. To size
+  /// the view, the selected size must be provided to the [render] method and it
+  /// must satisfy the constraints.
+  ///
+  /// When this changes, [PlatformDispatcher.onMetricsChanged] is called.
+  ///
+  /// At startup, the constraints for the view may not be known before Dart code
+  /// runs. If this value is observed early in the application lifecycle, it may
+  /// report constraints with all dimensions set to zero.
+  ///
+  /// This value does not take into account any on-screen keyboards or other
+  /// system UI. If the constraints are tight, the [padding] and [viewInsets]
+  /// properties provide information about how much of each side of the view may
+  /// be obscured by system UI. If the constraints are loose, this information
+  /// is not known upfront.
```

Comment:
What do you mean by "this information is not known"? Shall we just assert that only tight constraints can have non-zero padding and insets?

Reply:
This is basically describing platform limitations. No platform currently makes it possible to know this information upfront, but in theory a platform could provide it.